internal/cryptosigner/aws/kms.go

```diff
@@ -0,0 +1,164 @@
+package awssigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+)
+
+var (
+	ErrInvalidKeyAlgorithm = fmt.Errorf(`invalid key algorithm`)
+	ErrInvalidKeyID        = fmt.Errorf(`invalid key ID`)
+)
+
+type KMS struct {
+	alg    types.SigningAlgorithmSpec
+	jwaAlg jwa.KeyAlgorithm
+	client *kms.Client
+	kid    string
+}
+
+// NewKMS creates a new ECDSA object. This object isnot complete by itself -- it
+// needs is setup with the algorithm name to use (see
+// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/kms/types#SigningAlgorithmSpec),
+// a key ID to use while the AWS SDK makes network
+// requests.
+
+// NewKMS creates a new crypto signer which uses AWS KMS to sign data. The keys signing algorithm spec
+// dictates the type of signature that will be generated (see
+// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/kms/types#SigningAlgorithmSpec),
+
+// The key ID is the unique identifier of the KMS key or key alias.
+func NewKMS(client *kms.Client, kmsKeyID string) (*KMS, error) {
+	if kmsKeyID == "" {
+		return nil, ErrInvalidKeyID
+	}
+
+	keyDesc, err := client.GetPublicKey(context.Background(), &kms.GetPublicKeyInput{KeyId: aws.String(kmsKeyID)})
+	if err != nil {
+		return nil, fmt.Errorf(`failed to describe key %q: %w`, kmsKeyID, err)
```

Comment:
We began preferring `"`-strings where possible instead of backtick-strings for most uses a while ago. If nothing else, this file is inconsistent with the rest of the PR.

```suggestion
		return nil, fmt.Errorf("failed to describe key %q: %w", kmsKeyID, err)
```